spec/requests/placements/schools/mentors/add_mentor_spec.rb

```diff
@@ -2,14 +2,15 @@
 
 RSpec.describe "'Add mentor' journey", service: :placements, type: :request do
   let(:current_user) { create(:placements_user, schools: [school]) }
-  let(:start_path) { new_add_mentor_placements_school_mentors_path(school_id:) }
+  let!(:state_key) { SecureRandom.uuid }
+  let(:start_path) { new_add_mentor_placements_school_mentors_path(state_key:, school_id:) }
```

Comment:
The `/new` path doesn't have a `state_key` route param. We shouldn't need to provide one – `/new` will redirect us to `/edit/:state_key/name-of-first-step`, where `:state_key` has been generated automatically by the controller.

```suggestion
  let(:start_path) { new_add_mentor_placements_school_mentors_path(school_id:) }
```

Since `state_key` is not a route param, I expect it probably becomes a query string param when provided to this helper. For example:

```
/schools/:school_id/placements/new?state_key=03e173a4-2266-405e-8427-0410921f8ced
```

In fact, I'm wondering if we should also be altering this test to make sure it generates a new state key and redirects the user to it. What do you think? 🤔

https://github.com/DFE-Digital/itt-mentor-services/blob/c7a56194a1cc3e249e967a2456aabb407b5afba9/spec/support/shared_examples/requests/placements/add_a_mentor.rb#L7-L22